test/pluginSpec.js

```diff
@@ -31,8 +38,99 @@ describe('pluginSpec', () => {
+        protocolName: 'info',
+        contentType: BtpPacket.MIME_APPLICATION_JSON,
+        data: Buffer.from(JSON.stringify({
+          currencyScale: 8
+        }))
+      }]})
+
+      await assert.isRejected(
+        this.plugin._connect(),
+        /Fatal! Currency scale mismatch/)
+    })
+  })
+
   describe('sendMoney', () => {
-    it('verifies signature of last claim', () => {
+    describe('with high scale', function () {
+      beforeEach(function () {
+        this.opts.currencyScale = 9
+        this.plugin = new Plugin(this.opts)
+
+        this.sinon.stub(nacl.sign, 'detached').returns('abcdef')
+
+        this.plugin._paychan = this.plugin._channelDetails = { amount: '10', balance: '1' }
+        this.plugin._keyPair = {}
+        this.plugin._funding = true
+        this.plugin._channel = 'abcdef'
+        this.plugin._clientChannel = 'abcdef'
+        this.plugin._bestClaim = this.plugin._lastClaim = {
+          amount: '990',
+          signature: 'abcdef'
+        }
+      })
+
+      it('should round high-scale amount up to next drop', async function () {
+        const encodeSpy = this.sinon.spy(util, 'encodeClaim')
+        this.sinon.stub(this.plugin, '_call').resolves(null)
+
+        await this.plugin.sendMoney(100)
+
+        assert.deepEqual(encodeSpy.getCall(0).args, [ '1', 'abcdef' ])
+        assert.deepEqual(encodeSpy.getCall(1).args, [ '2', 'abcdef' ])
+      })
+
+      it('should keep error under a drop even on repeated roundings', async function () {
+        const encodeSpy = this.sinon.spy(util, 'encodeClaim')
+        this.sinon.stub(this.plugin, '_call').resolves(null)
+
+        await this.plugin.sendMoney(100)
+
+        // we can't stub verify for some reason so we need to prevent
+        // signature verification from happening
+        this.plugin._channelDetails.balance = '2'
+
+        await this.plugin.sendMoney(100)
+
+        assert.deepEqual(encodeSpy.getCall(0).args, [ '1', 'abcdef' ])
+        assert.deepEqual(encodeSpy.getCall(1).args, [ '2', 'abcdef' ])
+        assert.deepEqual(encodeSpy.getCall(2).args, [ '2', 'abcdef' ])
+        assert.deepEqual(encodeSpy.getCall(3).args, [ '2', 'abcdef' ])
+      })
+
+      it('should handle a claim', async function () {
+        // this stub isn't working, which is why handleMoney is throwing
+        this.sinon.stub(nacl.sign.detached, 'verify').returns('abcdef')
+        const encodeSpy = this.sinon.spy(util, 'encodeClaim')
+
+        await assert.isRejected(this.plugin._handleMoney(null, {
+          requestId: 1,
+          data: {
+            amount: '160',
+            protocolData: [{
+              protocolName: 'claim',
+              contentType: BtpPacket.MIME_APPLICATION_JSON,
+              data: Buffer.from(JSON.stringify({
+                amount: '1150',
+                signature: 'abcdef'
+              }))
+            }]
+          }
+        }), /Invalid claim signature/)
```

Comment:
Are you sure this test is working as intended? Which scenario is this testing?

Reply:
I couldn't stub out the `nacl.sign.detached.verify` function for some reason. So the signature verification fails, but when I check the args with which `encodeStub` was called I can tell that it was rounding the amount correctly.

Reply:
good catch, looks like the error here was because it was trying to take substring of the public key, which was undefined. The test case is just making sure that the encode call was called with a correctly rounded amount, though.